components/GotApi.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from "react";
+import {
+	Text,
+	ActivityIndicator,
+	ScrollView,
+	TouchableOpacity,
+	StyleSheet,
+} from "react-native";
+import { Accelerometer } from "expo-sensors";
+
+const GotApi = () => {
+	const [quote, setQuote] = useState({});
+	const [loading, setLoading] = useState(false);
+
+	//communication between code and position of phone
+	const [subscription, setSubscription] = useState(null);
+	const [data, setData] = useState({
+		x: 0,
+		y: 0,
+		z: 0,
+	});
+
+	//generates quote when button is clicked
+	useEffect(() => {
```

Comment:
Why having it here? When you have the fetch below - you should just be able to call the function in your return (which you do on line 78) without having to use useEffects once again up here.. maybe this is why your loader doesn't work..?